go/libraries/doltcore/sqle/enginetest/dolt_transaction_commit_test.go

```diff
@@ -456,3 +456,244 @@ func TestDoltTransactionCommitLateFkResolution(t *testing.T) {
 		},
 	})
 }
+
+var internalMergeScripts = []queries.TransactionTest{
+	{
+		Name: "Internal merges that produce a constraint violation with dolt_force_transaction_commit = off should be rolled back.",
+		SetUpScript: []string{
+			"CREATE TABLE t (pk int PRIMARY KEY, col1 int UNIQUE);",
+			"CALL DOLT_COMMIT('-am', 'create table');",
+
+			"CALL DOLT_CHECKOUT('-b', 'right');",
+			"INSERT INTO t values (2, 1);",
+			"CALL DOLT_COMMIT('-am', 'right edit');",
+
+			"CALL DOLT_CHECKOUT('main');",
+			"INSERT INTO t VALUES (1, 1);",
+			"CALL DOLT_COMMIT('-am', 'left edit');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query: "/* client a */ CALL DOLT_MERGE('right');",
+				ExpectedErrStr: "Committing this transaction resulted in a working set with constraint violations, transaction rolled back. " +
+					"This constraint violation may be the result of a previous merge or the result of transaction sequencing. " +
+					"Constraint violations from a merge can be resolved using the dolt_constraint_violations table before committing the transaction. " +
+					"To allow transactions to be committed with constraint violations from a merge or transaction sequencing set @@dolt_force_transaction_commit=1.",
+			},
+			{
+				Query:    "/* client a */ SELECT * from DOLT_CONSTRAINT_VIOLATIONS;",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:          "/* client a */ CALL DOLT_MERGE('--abort');",
+				ExpectedErrStr: "fatal: There is no merge to abort",
+			},
+		},
+	},
+	{
+		Name: "Internal merges that produce a conflict with dolt_allow_commit_conflicts = off should be rolled back.",
+		SetUpScript: []string{
+			"CREATE TABLE t (pk int PRIMARY KEY, col1 int);",
+			"CALL DOLT_COMMIT('-am', 'create table');",
+
+			"CALL DOLT_CHECKOUT('-b', 'right');",
+			"INSERT INTO t values (1, 100);",
+			"CALL DOLT_COMMIT('-am', 'right edit');",
+
+			"CALL DOLT_CHECKOUT('main');",
+			"INSERT INTO t VALUES (1, 200);",
+			"CALL DOLT_COMMIT('-am', 'left edit');",
+		},
+		Assertions: []queries.ScriptTestAssertion{
+			{
+				Query:          "/* client a */ CALL DOLT_MERGE('right');",
+				ExpectedErrStr: "Merge conflict detected, transaction rolled back. Merge conflicts must be resolved using the dolt_conflicts tables before committing a transaction. To commit transactions with merge conflicts, set @@dolt_allow_commit_conflicts = 1",
+			},
+			{
+				Query:    "/* client a */ SELECT * from DOLT_CONSTRAINT_VIOLATIONS;",
+				Expected: []sql.Row{},
+			},
+			{
+				Query:          "/* client a */ CALL DOLT_MERGE('--abort');",
+				ExpectedErrStr: "fatal: There is no merge to abort",
+			},
+		},
+	},
+}
+
+// Tests behavior of internal merges ( merges that occur inside a transaction ).
+func TestInternalMerges(t *testing.T) {
+	harness := newDoltHarness(t)
+	for _, script := range internalMergeScripts {
+		enginetest.TestTransactionScript(t, harness, script)
+	}
+}
+
+var transactionMergeScripts = []queries.TransactionTest{
```

Comment:
Same comment about test organization

I'm fine with having multiple blocks of tests that test different themes, but I don't know that this organizational scheme makes sense to me